effect/default_effector_stream.go

```diff
@@ -0,0 +1,69 @@
+package effect
+
+// DefaultEffectorStream is the default implementation of interface EffectorStream.
+type DefaultEffectorStream struct {
+	done bool
+	res bool
+	expr string
+	idx int
+	cap int
+	expl []int
+}
+
+func pushIndex(truthValue bool, s *[]int, index int) {
+	if truthValue {
+		*s = append(*s, index)
+	}
+}
+
+func (s *DefaultEffectorStream) Next() bool {
+	if !s.done {
+		panic("done should be true")
+	}
+	return s.res
+}
+
+func (s *DefaultEffectorStream) Explain() []int {
```

Comment:
Sorry, I missed the Explain. There should be only one policy that affects enforce result.

What do you think @DivyPatel9881 ?

When `res == true && done ==true` set the hit policy index to `expl`. `Explain` returns the index then.

Reply:
@nodece  `some(where (p_eft == allow)) && !some(where (p_eft == deny))` can have multiple policies hitted